lib/stm32/common/adc_common_v2.c

```diff
@@ -147,6 +147,37 @@ void adc_power_off(uint32_t adc)
 }
 
 /**
+ * Start the ADC calibration and immediately return.
+ * @sa adc_calibrate
+ * @sa adc_is_calibrate
+ * @param adc ADC Block register address base @ref adc_reg_base
+ */
+void adc_calibrate_async(uint32_t adc)
+{
+	ADC_CR(adc) = ADC_CR_ADCAL;
+}
+
+/**
+ * Is the ADC Calibrating?
+ * @param adc ADC Block register address base @ref adc_reg_base
+ * @return true if the adc is currently calibrating
+ */
+bool adc_is_calibrate(uint32_t adc)
```

Comment:
absolutely, I had that first.  I was just thinking of leaving it with the same word "calibrate" all the time, but yeah, I think my first choice, "is_calibrating" is indeed better now :)